src/static/riot/competitions/detail/submission_upload.tag

```diff
@@ -156,9 +156,17 @@
                 let event_data = JSON.parse(event.data)
                 switch (event_data.type) {
                     case 'catchup':
+                        let detailed_result_url = ''
                         _.forEach(_.compact(event_data.data.split('\n')), data => {
-                            self.handle_websocket(event_data.submission_id, JSON.parse(data))
+                            data = JSON.parse(data)
+                            if (data.kind === 'detailed_result_update') {
+                                detailed_result_url = data.result_url
+                            } else {
+                                self.handle_websocket(event_data.submission_id, JSON.parse(data))
```

Comment:
parsing data twice here?

Reply:
Good catch.

Reply:
little things like this, we should get in the habit of using the suggestion interface in github, you can batch commit all the suggestions and save time opening pycharm to make changes.

Reply:
```suggestion
                                self.handle_websocket(event_data.submission_id, data)
```